ksql-engine/src/main/java/io/confluent/ksql/function/UdfTemplate.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License; you may not use this file
+ * except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function;
+
+import com.google.common.primitives.Primitives;
+import com.squareup.javapoet.CodeBlock;
+import java.lang.reflect.Method;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@SuppressWarnings("WeakerAccess") // used from generated code
+public final class UdfTemplate {
+
+  private static final Logger LOG = LoggerFactory.getLogger(UdfTemplate.class);
+
+  private UdfTemplate() { /* private constructor for utility class */ }
+
+  public static String generateCode(final Method method, final String obj) {
+    final Class<?>[] params = method.getParameterTypes();
+
+    final CodeBlock.Builder code = CodeBlock.builder();
+    for (int idx = 0; idx < params.length; idx++) {
+      final Class<?> param = params[idx];
+      code.addStatement("$T arg$L = ($T) $T.coerce(args, $T.class, $L)",
+                        Primitives.wrap(param),
+                        idx,
+                        Primitives.wrap(param),
+                        UdfTemplate.class,
+                        param,
+                        idx);
+    }
+
+    final String args = IntStream.range(0, params.length)
+        .mapToObj(i -> "arg" + i)
+        .collect(Collectors.joining(", "));
+
+    code.addStatement("return (($T) $L).$L($L)",
+                      method.getDeclaringClass(), obj, method.getName(), args);
+
+    final String codeString = code.build().toString();
+    LOG.trace("Generated code:\n" + codeString);
+    return codeString;
+  }
+
+  @SuppressWarnings({"unchecked"}) // used by code generator
```

Comment:
nit: not sure how 'used by code generator' explains why `unchecked` warnings can be disabled... ;)

Reply:
derp 🤕